### PR TITLE
draft for #2815, #2816

### PR DESCRIPTION
**Pull request by Kantis**
@sksamuel is this something in line with what you were thinking? Lots of work remain, mainly curious about what you think of the general idea.

I yoinked the code I was playing around with for making JSON extractions, think it could be useful here..

### CODE REVIEW

**Comment by sksamuel** [comment]
@Kantis this ready for review ?

File: kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import io.kotest.assertions.json.pretty
```
Comment (LeoColman) [inline kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt:1]: I think this is missing a package statement, isn't it?

File: kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   while (i < path.length - 1) {
+      when (path[i]) {
+         ']' -> i++
+         '.' -> jsonExtractions.add(ByProperty(path.substring(i + 1, nextToken())))
+         '[' -> jsonExtractions.add(ByIndexing(path.substring(i + 1, nextToken()).toInt()))
+         else -> error("Invalid input $path. (i = $i, actions = $jsonExtractions)")
+      }
+   }
+
+   println("Returning actions: $jsonExtractions")
```
Comment (LeoColman) [inline kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt:92]: This seems like a hanging debug println.
Is it supposed to be here?

File: kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt (hunk 1 above)
Comment (Kantis) [reply kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt:1]: Yup, you're right. Will fix :)

File: kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt (hunk 2 above)
Comment (Kantis) [reply kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/array.kt:92]: Indeed, will remove.

**Comment by sksamuel** [comment]
@Kantis do you think this is similar to https://github.com/kotest/kotest/pull/2861

**Comment by Kantis** [comment]
> @Kantis do you think this is similar to #2861

Yeah, very. 😁 I could add the path-matching capabilities on top of that PR's work instead, if we're at all interested in it. 

Either way, I will close this.

**Pull request closed without merging**